extensions/wikia/TemplateClassification/TemplateClassificationMockApiController.class.php

```diff
@@ -6,8 +6,24 @@
  */
 
 class TemplateClassificationMockApiController extends WikiaApiController {
+	const TYPE_FIELD = 'type';
 
 	public function getTemplateType() {
-		$this->setVal( 'type', array_rand( TemplateClassification::$templateTypes ) );
+		$articleId = $this->request->getInt( 'articleId' );
+		/* Use cache to have consistent results for same pages due to randomization */
+		$type = WikiaDataAccess::cache(
+			wfMemcKey( 'template-classification-type-for-page', $articleId ),
+			WikiaResponse::CACHE_STANDARD,
+			function () {
+				$rand = array_rand( TemplateClassification::$templateTypes );
+				return TemplateClassification::$templateTypes[$rand];
+			}
+		);
+		$this->response->setVal( self::TYPE_FIELD, $type );
+	}
+
+	public function setTemplateType() {
```

Comment:
For the sake of testing, you could add another parameter that would imply a success or failure of a request. :)